python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -73,6 +73,10 @@ def __init__(self, options: ExecutionOptions):
         self._topology: Optional[Topology] = None
         self._output_node: Optional[OpState] = None
 
+        self._dataset_metrics = _get_or_create_dataset_metrics()
+        self._prev_metrics_state = {}
```

Comment:
Store previous metrics state because metrics like `bytes_spilled` we need to report the change in bytes.

Reply:
Not needed anymore since everything is a gauge